src/accountant.rs

```diff
@@ -164,17 +177,25 @@ impl Accountant {
 
     /// Process a Transaction that has already been verified.
     pub fn process_verified_transaction(&self, tr: &Transaction) -> Result<()> {
-        self.process_verified_transaction_debits(tr)?;
-        self.process_verified_transaction_credits(tr);
-        Ok(())
+        return match self.process_verified_transaction_debits(tr) {
+            Ok(_) => {
+                self.process_verified_transaction_credits(tr);
+                Ok(())
+            }
+            Err(err) => Err(err),
+        };
     }
 
     /// Process a batch of verified transactions.
     pub fn process_verified_transactions(&self, trs: Vec<Transaction>) -> Vec<Result<Transaction>> {
         // Run all debits first to filter out any transactions that can't be processed
         // in parallel deterministically.
         let results: Vec<_> = trs.into_par_iter()
-            .map(|tr| self.process_verified_transaction_debits(&tr).map(|_| tr))
+            .filter_map(|tr| match self.process_verified_transaction_debits(&tr) {
```

Comment:
This is probably a slower version of the original code that loses information in those `Err` values.

Reply:
This is unnecessary now since we're looping on failed transactions, but I think it will raise errors if you have transactions that fail in `process_packets_bench` since they're not being filtered out